engine/source/output/h3d/h3d_build_cpp/c_h3d_create_nodal_scalar_datatype.cpp

```diff
@@ -69,20 +69,25 @@ extern "C"
 /*=================================================================*/
 
 void c_h3d_create_nodal_scalar_datatype_(int *cpt_data, char *name, int *size, int *info, char *name1, int *size1, int *INTER_ID,
-                                         char *comment, int *s_comment)
+                                         char *comment, int *s_comment, int *is_inter, int *size_inter0, char *ch_inter0)
 {
-    char *cname,*cname1,*ccomment;
-    int cname_len,cname_len1,ccomment_len;
+    char *cname,*cname1,*ccomment,*chinter;
+    int cname_len,cname_len1,ccomment_len,chinter_len;
     int i;
     float node[3]; 
     H3D_ID node_id;
     unsigned int elem_count = 1;
     char* edata_type;
     
     char * CH_INTER_ID = new char [100];
-    CH_INTER_ID[0] ='\0'; 
+    CH_INTER_ID[0] ='\0';  
 
-    cname_len = *size + 10;
+    chinter_len = 10 + *size_inter0;
+    chinter=(char*) malloc(sizeof(char)*chinter_len);
```

Comment:
Memory leak. Not freed.

Reply:
Hi @laurent-altr  can you explain please the problem you found

Reply:
Hi @MahaHachani , you need after an array malloc to make somewhere a free(array) in order not to have memory leaks .